awslambdaric/lambda_runtime_client.py

```diff
@@ -68,10 +67,25 @@ def post_init_error(self, error_response_data):
         if response.code != http.HTTPStatus.ACCEPTED:
             raise LambdaRuntimeClientError(endpoint, response.code, response_body)
 
-    def wait_next_invocation(self):
-        with ThreadPoolExecutor() as e:
-            fut = e.submit(runtime_client.next)
-        response_body, headers = fut.result()
+    def wait_next_invocation(self, use_thread_for_polling_next=False):
+        # Calling runtime_client.next() from a separate thread unblocks the main thread,
+        # which can then process signals.
+        if use_thread_for_polling_next:
+            from concurrent.futures import ThreadPoolExecutor
+            from .lambda_runtime_exception import FaultException
```

Comment:
Let's move this to line 82

Reply:
It doesn't work. Hence reverted back to as it was.

Reply:
Let's move it back to the top of the file since it's already not a lazy import in bootstrap.py